ext/reflection/php_reflection.c

```diff
@@ -2223,34 +2226,24 @@ ZEND_METHOD(ReflectionParameter, __construct)
 	if (fptr->common.fn_flags & ZEND_ACC_VARIADIC) {
 		num_args++;
 	}
-	if (Z_TYPE_P(parameter) == IS_LONG) {
-		position= (int)Z_LVAL_P(parameter);
-		if (position < 0 || (uint32_t)position >= num_args) {
-			_DO_THROW("The parameter specified by its offset could not be found");
-			goto failure;
-		}
-	} else {
+	if (string_parameter != NULL) {
 		uint32_t i;
 
 		position = -1;
-		if (!try_convert_to_string(parameter)) {
-			goto failure;
-		}
 
 		if (has_internal_arg_info(fptr)) {
 			for (i = 0; i < num_args; i++) {
 				if (arg_info[i].name) {
-					if (strcmp(((zend_internal_arg_info*)arg_info)[i].name, Z_STRVAL_P(parameter)) == 0) {
+					if (zend_string_equals_literal(string_parameter, ((zend_internal_arg_info*)arg_info)[i].name)) {
```

Comment:
This function does need a literal, this is going to use the wrong size. Probably not tested well enough...

Reply:
But I'm comparing it to a  literal, aren't I? Or am I misunderstanding what needs to be compared to a literal?